buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/buildtypes/BuildTypesPlugin.kt

```diff
@@ -94,16 +95,43 @@ fun Project.insertBuildTypeTasksInto(
     fun forEachBuildTypeTask(act: (String) -> Unit) =
         buildType.tasks.reversed().forEach(act)
 
+    fun ensureBuildTypeTaskOrdering(matchingTasks: List<Task>) = {
+        matchingTasks.forEach { t ->
+            taskList.forEach {
+                t.shouldRunAfter(it)
+            }
+        }
+    }
+
     when {
         subproject.isEmpty() ->
-            forEachBuildTypeTask(::insert)
+            forEachBuildTypeTask {
+                val matchingTasks = ArrayList<Task>()
+                getAllprojects().forEach { p ->
+                    val matchingTask = p.getTasks().findByName(it)
+                    if (matchingTask != null) {
+                        buildTypeTask.configure {
+                            dependsOn(matchingTask)
```

Comment:
I don't quite understand this. Let's see we're running `./gradlew quickTest`. The `taskList` is empty because 

```
afterEvaluate {
  usedTaskNames.forEach { (index, usedName) ->
    invokedTaskNames.removeAt(index) // all requested build types have already been removed here
```

So `quickTest` will not be in final running tasks. What's the point of making it depend on other tasks? Am I missing anything?